### PR TITLE
feat: verification from SDK

### PR DESCRIPTION
**Pull request by cristianIOHK**
### Description: 
This PR is for ATL-6512 andadds the logic required to initiate a verification request from the SDK. This allows a holder to holder verification, respond to that request with a presentation submission and validate the presentation received.

### Checklist: 
- [x] My PR follows the contribution guidelines of this project
- [x] My PR is free of third-party dependencies that don't comply with the [Allowlist](https://toc.hyperledger.org/governing-documents/allowed-third-party-license-policy.html#approved-licenses-for-allowlist)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked the PR title to follow the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: edge-agent-sdk/src/androidInstrumentedTest/kotlin/org/hyperledger/identus/walletsdk/prismagent/AnoncredsTests.kt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 @file:Suppress("ktlint:standard:import-ordering")
 
-package org.hyperledger.identus.walletsdk.edgeagent
+package org.hyperledger.identus.walletsdk.prismagent
```
Comment (elribonazo) [inline edge-agent-sdk/src/androidInstrumentedTest/kotlin/org/hyperledger/identus/walletsdk/prismagent/AnoncredsTests.kt:3]: Why prismagent again aren't we running away from this?

File: sampleapp/src/main/res/xml/network_security_config.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <network-security-config>
     <domain-config cleartextTrafficPermitted="true">
-        <domain includeSubdomains="true">192.168.68.103</domain>
+        <domain includeSubdomains="true">192.168.68.114</domain>
```
Comment (elribonazo) [inline sampleapp/src/main/res/xml/network_security_config.xml:4]: Would propose to set

```
<domain includeSubdomains="true">192.168.*.*</domain> 
```

Using wildcard instead which basically will allow any of your local IP addresses and most of us will use the same ranges by default so this covers most cases and u don't need to change everytime your DHCP assigns you an IP

**Review by elribonazo — CHANGES_REQUESTED** [review]

File: edge-agent-sdk/src/androidInstrumentedTest/kotlin/org/hyperledger/identus/walletsdk/prismagent/AnoncredsTests.kt [hunk 3]
```diff
@@ -24,6 +24,9 @@ import junit.framework.TestCase.assertTrue
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.test.runTest
 import kotlinx.serialization.json.Json
+import org.hyperledger.identus.walletsdk.edgeagent.ConnectionManager
+import org.hyperledger.identus.walletsdk.edgeagent.ConnectionManagerImpl
+import org.hyperledger.identus.walletsdk.edgeagent.PrismAgent
```
[the remark this answers is not in the document]
Comment (cristianIOHK) [reply edge-agent-sdk/src/androidInstrumentedTest/kotlin/org/hyperledger/identus/walletsdk/prismagent/AnoncredsTests.kt:29]: Done

File: edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/castor/CastorImpl.kt [hunk 4]
```diff
@@ -131,7 +147,7 @@ constructor(
     ): Boolean {
         val document = resolveDID(did.toString())
         val publicKeys: List<PublicKey> =
-            CastorShared.getKeyPairFromCoreProperties(document.coreProperties)
+            getPublicKeysFromCoreProperties(document.coreProperties)
 
         if (publicKeys.isEmpty()) {
             throw CastorError.InvalidKeyError("KeyPairs is empty")
```
[the remark this answers is not in the document]
Comment (cristianIOHK) [reply edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/castor/CastorImpl.kt:153]: Good point.

File: sampleapp/src/main/res/xml/network_security_config.xml (hunk 2 above)
Comment (cristianIOHK) [reply sampleapp/src/main/res/xml/network_security_config.xml:4]: I tried that and does not work unfortunately.

**Review by elribonazo — APPROVED** [review]

**Pull request merged**